File: kaleido-gradle-plugin/src/main/java/com/tongsr/kaleido/gradle/dsl/KaleidoResources.java
package com.tongsr.kaleido.gradle.dsl;

import org.gradle.api.file.RegularFileProperty;
import org.gradle.api.provider.Property;
import org.gradle.api.provider.SetProperty;

public abstract class KaleidoResources {
    public abstract SetProperty<String> getNativeLibrariesToDelete();

    public abstract SetProperty<String> getMetadataToDelete();

    public abstract Property<Boolean> getReplaceUnusedStrings();

    public abstract RegularFileProperty getConfirmedUnusedStringsFile();

    public abstract SetProperty<String> getRetainedLanguages();
}

File: kaleido-gradle-plugin/src/main/kotlin/com/tongsr/kaleido/gradle/dsl/KaleidoResources.kt
package com.tongsr.kaleido.gradle.dsl

import org.gradle.api.file.RegularFileProperty
import org.gradle.api.provider.Property
import org.gradle.api.provider.SetProperty

abstract class KaleidoResources {
    abstract val nativeLibrariesToDelete: SetProperty<String>
    abstract val metadataToDelete: SetProperty<String>
    abstract val replaceUnusedStrings: Property<Boolean>
    abstract val confirmedUnusedStringsFile: RegularFileProperty
    abstract val retainedLanguages: SetProperty<String>
}
